presto-main/src/main/java/com/facebook/presto/sql/planner/RowExpressionInterpreter.java

```diff
@@ -292,6 +295,11 @@ public Object visitLambda(LambdaDefinitionExpression node, Object context)
             if (optimizationLevel.ordinal() < EVALUATED.ordinal()) {
                 // TODO: enable optimization related to lambda expression
                 // Currently, we are not able to determine if lambda is deterministic.
+                // context is passed down as null here since lambda argument can only be resolved under the evaluation context.
```

Comment:
Maybe we can now remove the `TODO` comment above since we implement it?

Reply:
We are still not evaluating lambda (only have optimized its body) yet :) Lambda functions was treated as non-deterministic.